src/parentheses/Solution.java

```diff
@@ -0,0 +1,46 @@
+package parentheses;
+
+import java.util.Stack;
+
+public class Solution {
+    public boolean isValid(String parentheses) {
+
+        char[] arrayParentheses = parentheses.toCharArray();
+        Stack<Character> stack = new Stack<>();
+        boolean areValidParentheses = true;
+        for (int i = 0; i < arrayParentheses.length; i++) {
+            char character = arrayParentheses[i];
+            boolean isOpencharacter = character == '[' || character == '{' || character == '(';
+
+            if (isOpencharacter) {
+                stack.push(character);
+                System.out.println("stack: " + stack);
+            } else {
+                switch (character) {
+                    case ')':
+                        areValidParentheses = areParenthesesBalanced(stack, '(');
+                        break;
+                    case '}':
+                        areValidParentheses = areParenthesesBalanced(stack, '{');
+                        break;
+                    case ']':
+                        areValidParentheses = areParenthesesBalanced(stack, '[');
+                        break;
+                    default:
+                        throw new IllegalStateException("Unexpected value: " + character);
+                }
+            }
+        }
+        return areValidParentheses;
+    }
+
+    private boolean areParenthesesBalanced(Stack<Character> stack, char openCharacter) {
+        boolean isCloseParenthesesCorrect = true;
+        if (stack.isEmpty() || stack.pop() != openCharacter) {
+            isCloseParenthesesCorrect = false;
+        }
+        return isCloseParenthesesCorrect;
+    }
+}
+
+
```

Comment:
Please only have one blank line at the end of the file. Here we have two.

Reply:
Thanks, I will leave only one blank line.